src/app/games/sahara/SaharaMove.ts

```diff
@@ -29,38 +32,48 @@ export class SaharaMove extends MoveCoordToCoord {
             encodedMove -= sy;
             encodedMove /= 6;
             const sx: number = encodedMove;
-            return new SaharaMove(new Coord(sx, sy), new Coord(ex, ey));
+            return SaharaMove.from(new Coord(sx, sy), new Coord(ex, ey)).get();
         }
     }
-    public static checkDistanceAndLocation(start: Coord, end: Coord): void {
+    public static checkDistanceAndLocation(start: Coord, end: Coord): MGPValidation {
         const dx: number = Math.abs(start.x - end.x);
         const dy: number = Math.abs(start.y - end.y);
         const distance: number = dx+dy;
-        if (distance === 1) {
-            const fakeNeighboors: Coord = TriangularCheckerBoard.getFakeNeighboors(start);
-            if (end.equals(fakeNeighboors)) {
-                throw new Error(start.toString() + ' and ' + end.toString() + ' are not neighboors.');
+        if (distance === 0) {
+            return MGPValidation.failure(RulesFailure.MOVE_CANNOT_BE_STATIC());
+        } else if (distance === 1) {
+            const fakeNeighbors: Coord = TriangularCheckerBoard.getFakeNeighbors(start);
+            if (end.equals(fakeNeighbors)) {
+                return MGPValidation.failure(SaharaFailure.THOSES_TWO_SPACE_ARE_NOT_NEIGHBORS());
             }
         } else if (distance === 2) {
             if ((start.x + start.y) % 2 === 0) {
-                throw new Error(SaharaFailure.CAN_ONLY_REBOUND_ON_BLACK());
+                return MGPValidation.failure(SaharaFailure.CAN_ONLY_REBOUND_ON_BLACK());
             }
             if (start.x === end.x) {
-                throw new Error(start.toString() + ' and ' + end.toString() + ' have no intermediary neighboors.');
+                return MGPValidation.failure($localize`Thoses two spaces have no intermediary neighbor.`);
             }
         } else {
-            throw new Error($localize`You can move one or two spaces, not ${distance}.`);
+            return MGPValidation.failure($localize`You can move one or two spaces, not ${distance}.`);
         }
+        return MGPValidation.SUCCESS;
     }
-    constructor(start: Coord, end: Coord) {
-        super(start, end);
+    public static from(start: Coord, end: Coord): MGPFallible<SaharaMove> {
         if (!start.isInRange(SaharaState.WIDTH, SaharaState.HEIGHT)) {
             throw new Error('Move must start inside the board not at ' + start.toString() + '.');
```

Comment:
use MGPFallible instead of throwing ?

Reply:
J'y pensais, mais ça c'est quelque chose que j'estime qui ne devrait juste pas arriver, contrairement aux autres qui sont "normal clic related", du coup, ça obligerait à i18ner des code qui n'arriveront jamais (c'qui serait polluer les todolist de nos futur 25 interprètes quand on sera fully internationaux :D ! (non j'plaisante, c'est surtout ~sale ?))